pallet/sylo/src/lib.rs

```diff
@@ -0,0 +1,459 @@
+			id: Vec<u8>,
+			controller: T::AccountId,
+			service_endpoints: BoundedVec<BoundedVec<u8, T::StringLimit>, T::MaxServiceEndpoints>,
+		},
+		ResolverUnregistered {
+			id: Vec<u8>,
+		},
+		ValidationRecordCreated {
+			author: T::AccountId,
+			id: Vec<u8>,
+		},
+		ValidationEntryAdded {
+			author: T::AccountId,
+			id: Vec<u8>,
+			checksum: H256,
+		},
+		ValidationRecordUpdated {
+			author: T::AccountId,
+			id: Vec<u8>,
+			resolvers: Option<Vec<Vec<u8>>>,
+			data_type: Option<Vec<u8>>,
+			tags: Option<Vec<Vec<u8>>>,
+		},
+		ValidationRecordDeleted {
+			author: T::AccountId,
+			id: Vec<u8>,
+		},
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::call_index(0)]
+		#[pallet::weight({
+			T::WeightInfo::set_payment_asset()
+		})]
+		pub fn set_payment_asset(origin: OriginFor<T>, payment_asset: AssetId) -> DispatchResult {
+			ensure_root(origin)?;
+			<SyloAssetId<T>>::put(payment_asset);
+			Ok(())
+		}
+
+		#[pallet::call_index(1)]
+		#[pallet::weight({
+			T::WeightInfo::set_sylo_resolver_method()
+		})]
+		pub fn set_sylo_resolver_method(
+			origin: OriginFor<T>,
+			resolver_method: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			<SyloResolverMethod<T>>::put(resolver_method);
+			Ok(())
+		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight({
+			T::WeightInfo::register_resolver(<T::StringLimit>::get(), <T::MaxServiceEndpoints>::get())
+		})]
+		pub fn register_resolver(
+			origin: OriginFor<T>,
+			identifier: BoundedVec<u8, T::StringLimit>,
+			service_endpoints: BoundedVec<ServiceEndpoint<T::StringLimit>, T::MaxServiceEndpoints>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			ensure!(
+				<Resolvers<T>>::get(identifier.clone()).is_none(),
+				Error::<T>::ResolverAlreadyRegistered
+			);
+
+			let resolver =
+				Resolver { controller: who.clone(), service_endpoints: service_endpoints.clone() };
+
+			<Resolvers<T>>::insert(identifier.clone(), resolver);
+
+			Self::deposit_event(Event::ResolverRegistered {
+				id: identifier.to_vec(),
+				controller: who,
+				service_endpoints,
+			});
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight({
+			T::WeightInfo::update_resolver(<T::MaxServiceEndpoints>::get())
+		})]
+		pub fn update_resolver(
+			origin: OriginFor<T>,
+			identifier: BoundedVec<u8, T::StringLimit>,
+			service_endpoints: BoundedVec<ServiceEndpoint<T::StringLimit>, T::MaxServiceEndpoints>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let mut resolver =
+				<Resolvers<T>>::get(identifier.clone()).ok_or(Error::<T>::ResolverNotRegistered)?;
+
+			ensure!(who == resolver.controller, Error::<T>::NotController);
+
+			resolver.service_endpoints = service_endpoints.clone();
+
+			<Resolvers<T>>::insert(identifier.clone(), resolver);
+
+			Self::deposit_event(Event::ResolverUpdated {
+				id: identifier.to_vec(),
+				controller: who,
+				service_endpoints,
+			});
+
+			Ok(())
```

Comment:
nit: instead of getting and inserting:

```suggestion
			<Resolvers<T>>::try_mutate(&identifier, |maybe_resolver| {
                let resolver = maybe_resolver.as_mut().ok_or(Error::<T>::ResolverNotRegistered)?;
                ensure!(who == resolver.controller, Error::<T>::NotController);
                resolver.service_endpoints = service_endpoints.clone();
                
                Self::deposit_event(Event::ResolverUpdated {
                    id: identifier.into_vec(),
                    controller: who,
                    service_endpoints,
                });
            Ok(())
        })
```